nupic/research/frameworks/dendrites/plotting_utils.py

```diff
@@ -0,0 +1,419 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+
+from .utils import (
+    dendrite_duty_cycle,
+    dendrite_overlap,
+    dendrite_overlap_matrix,
+    entropy,
+    mean_selected_activations,
+    percent_active_dendrites,
+)
+
+
+def plot_dendrite_activations(
+    dendrite_segments,
```

Comment:
Looks well documented and clean. Can you make wandb optional? (I'm not familiar enough with wandb to know which lines of code are specific to it.). I should be able to use it from Jupyter lab or just save to a file. Since you are returning a figure, the client can decide what to do with it.

Reply:
@subutai  The files here are not using wandb, and just plotting & returning a figure. That figure can then be either pasted to wandb or saved to disk. Since there's consideration for writing a mixin to handle logging of metrics, I propose we should leave saving files to that, and the functions here still return figures.

Reply:
Makes sense.